tracee-rules/engine/engine.go

```diff
@@ -80,7 +83,7 @@ func (engine *Engine) Start(done chan bool) {
 			}
 		}(s, c)
 	}
-	<-done
+	engine.consumeSources(done)
```

Comment:
If we want to use deferred close(c) the consumeSources must run in the same thread as Start()

If for some reason consumeSources must run in its own goroutine we should not use deferred close(c) but consumeSources must do that in the loop when it receives data from the done channel as suggested by @yanivagman 

That's at least my understanding of what is going on here. I tested and we no longer observer panic on ctrl + C and when running benchmarks. I cannot see DATA RACES anymore